tracing/tracing/src/main/java/io/helidon/tracing/TracerProviderHelper.java

```diff
@@ -58,7 +58,12 @@ private TracerProviderHelper() {
     }
 
     public static Optional<Span> currentSpan() {
-        return TRACER_PROVIDER.currentSpan();
+        /*
+        If a custom TracerProvider implementation indirectly tries to access the current span (for example, by triggering custom
+        logging that adds the current span to each message), then this method can be invoked before the static initializer has
+        completed and, therefore, before TRACER_PROVIDER is assigned.
+         */
+        return (TRACER_PROVIDER == null) ? Optional.empty() : TRACER_PROVIDER.currentSpan();
```

Comment:
I think that this is creating a bug. We should never reach this method within initalization.
If we do, then we should change the way we initialize, rather than checking if a constant field is null

Reply:
As I mentioned in our DMs, we could add a static boolean that indicates if the static initializer is in progress or not. Then `currentSpan` could check that boolean and, if true, return `Optional.empty()` right away rather than trying to use `TRACER_PROVIDER`. That might look slightly less odd than checking a supposedly `final` field for null, but it amounts to the same thing: checking whether the static initializer is in-progress or not. 

As we also discussed, I will also add code to throw exceptions from other methods that rely on `TRACER_PROVIDER` but are `void` or have non-`Optional` return types.